producer.py

```diff
@@ -49,7 +49,10 @@ async def main() -> None:
     for package in unresolved_packages:
         package_name = unresolved_packages[package].name
         package_version = unresolved_packages[package].version
-        sources = [unresolved_packages[package].index]
+
+        sources = []
+        if unresolved_packages[package].index:
+            sources = [unresolved_packages[package].index]
```

Comment:
Should we raise an exception here if `unresolved_packages[package].index` is `None`?

Reply:
This is the producer, in case of adviser you might not have the index, the investigator consumer will make sure to consider only indexes registered to Thoth. This is to avoid also 
`{"package_name": "black", "package_version": "*", "index_url": [null], "solver": "solver-rhel-8-py36",` and index_url can be empty list